### PR TITLE
make aggregation of empty GroupedDataFrame correct with AsTable

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/3217

### CODE REVIEW

**Comment by bkamins** [comment]
CI error on x86 is due to OOM.

**Comment by bkamins** [comment]
I have also updated documentation which also fixes #3214.

File: src/groupeddataframe/splitapplycombine.jl [hunk 1]
```diff
@@ -486,6 +486,23 @@ function _combine_process_pair_symbol(optional_i::Bool,
...
+        throw(ArgumentError("keys of the returned elements must be " *
+                            "`Symbol`s, strings or integers"))
+    end
+    if any(x -> !isequal(keys(x), kp1), res)
+        throw(ArgumentError("keys of the returned elements must be identical"))
+    end
+    outcols = [[x[n] for x in res] for n in kp1]
+    # make sure we only infer column names and types for empty res, but do not
+    # produce values that were generated when computing firstres
+    emptyres && foreach(empty!, outcols)
```
Comment (nalimilan) [inline src/groupeddataframe/splitapplycombine.jl:501]: This seems to differ from what the docstring says, as it doesn't mention that the result has zero rows. Not sure which behavior is best.

Comment (bkamins) [reply src/groupeddataframe/splitapplycombine.jl:501]: We already have the behavior (for the single column case) that we only infer column type from the call. I have clarified (hopefully) the docstring.

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**